core/dbt/config/profile.py

```diff
@@ -273,6 +278,11 @@ def render_profile(
         elif "target" in raw_profile:
             # render the target if it was parsed from yaml
             target_name = renderer.render_value(raw_profile["target"])
+        elif is_secondary and len(raw_profile["outputs"]) == 1:
+            # if we only have one target, we can infer the target name
+            # currently, this is only used for secondary profiles
+            target_name = next(iter(raw_profile["outputs"]))
+            fire_event(MissingProfileTarget(profile_name=profile_name, target_name=target_name))
```

Comment:
Why is this event necessary in this case?

Reply:
The event name is confusing, but the message is useful for debugging purposes to indicate that we inferred a `target_name`:

https://github.com/dbt-labs/dbt-core/blob/e60b41d9fabb72fccd5e94a8ceb2a4f413c6f161/core/dbt/events/types.py#L68-L73

I'll add a comment for the same.

Reply:
Added a comment here, let me know if any other changes are needed here.